components/Articles.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { useChannel } from "@ably-labs/react-hooks";
+import { formatDate } from "../lib/dates";
+import ArticlePreview from "./ArticlePreview";
+import styles from "../styles/Home.module.css";
+
+export default function Articles(props) {
+  let inputBox = null;
+  const defaultMessages = props.history || [];
+
+  const [headlineText, setHeadlineText] = useState("");
+  const [headlines, updateHeadlines] = useState(defaultMessages);
+  const [_, ably] = useChannel("[?rewind=3]headlines", (headline) => {
+    updateHeadlines((prev) => [...prev, headline]);
+  });
+
+  const headlineTextIsEmpty = headlineText.trim().length === 0;
+  const processedHeadlines = headlines.map((headline) => processMessage(headline, ably.auth.clientId));
```

Comment:
Instead of cleaning up the data as the elements are generated, we are cleaning the message content ahead of time instead. 
It makes it easier to understand/follow.